tgui/packages/tgui/interfaces/AirVendor.tsx

```diff
@@ -15,48 +16,73 @@ type AirVendorParams = {
   cash: number,
   cardname: string,
   bankMoney: number,
-
+  vend_type:string,
   holding: string,
   holding_pressure: number,
   min_pressure: number,
   max_pressure: number,
+  air_cost: number,
   fill_cost: number,
   target_pressure: number,
+  current_fill: number
 }
 
 const VendorSection = (_props, context) => {
   const { act, data } = useBackend<AirVendorParams>(context);
-  const { cash, bankMoney, fill_cost, target_pressure, min_pressure, max_pressure } = data;
+  const {
+    air_cost,
+    bankMoney,
+    cash,
+    current_fill,
+    fill_cost,
+    max_pressure,
+    min_pressure,
+    target_pressure,
+    vend_type,
+  } = data;
 
   const handleFillClick = () => act('o2_fill');
-  const handleChangePressure = (pressure) => act('o2_changepressure', { pressure: pressure });
+  const handleChangePressure = (pressure: number) => act('o2_changepressure', { pressure: pressure });
 
-  const canVend = fill_cost > 0 && (bankMoney > fill_cost || cash > fill_cost);
+  const isFree = !air_cost;
+  const canVend = isFree || (fill_cost > 0 && (bankMoney > fill_cost || cash > fill_cost));
 
   return (
-    <Section title={"Buy Oxygen!"}>
+    <Section title={`Buy ${vend_type}!`}>
       <LabeledList>
         <LabeledList.Item label="Cost">
           <Button
-            content={(<>{fill_cost || 0}⪽</>)}
+            content={isFree ? 'Free!' : `${fill_cost || 0}⪽`}
             color={canVend ? "green" : "grey"}
             disabled={!canVend}
             onClick={handleFillClick} />
         </LabeledList.Item>
         <LabeledList.Item label="Pressure">
-          <Button
-            onClick={() => handleChangePressure(min_pressure)}
-            content="Min" />
-          <NumberInput
-            animated
-            width={6}
-            value={target_pressure}
-            minValue={min_pressure}
-            maxValue={max_pressure}
-            onChange={(_e, new_pressure) => handleChangePressure(new_pressure)} />
-          <Button
-            onClick={() => handleChangePressure(max_pressure)}
-            content="Max" />
+          <Flex>
```

Comment:
Would `<Stack>` be appropriate here?
https://github.com/goonstation/goonstation/blob/master/tgui/docs/component-reference.md#stack

Reply:
implemented